lib/knapsack_pro/base_allocator_builder.rb

```diff
@@ -38,7 +38,27 @@ def test_file_pattern
     end
 
     def test_files
-      KnapsackPro::TestFileFinder.call(test_file_pattern)
+      found_test_files = KnapsackPro::TestFileFinder.call(test_file_pattern)
+
+      if adapter_class == KnapsackPro::Adapters::RSpecAdapter && KnapsackPro::Config::Env.rspec_split_by_test_examples?
+        test_files_count = found_test_files.size
+
+        KnapsackPro.logger.warn("Generating RSpec test examples JSON report to prepare your test suite to be split by test examples (by individual 'it'. Thanks to that a single test file can be split across parallel CI nodes). Analyzing #{test_files_count} test files.")
+
+        if test_files_count > 1000
+          KnapsackPro.logger.warn("You have more than 1000 test files, it may take longer to generate test examples. Please wait...")
+        end
+
+        # generate RSpec JSON report in separate process to not pollute RSpec state
+        cmd = 'bundle exec rake knapsack_pro:rspec_test_example_detector'
+        Kernel.system(cmd)
```

Comment:
I wonder if we should have a backup in case this fails. We could react here differently depending on the return value. (For reference, according to the docs: `true` means zero exit status, `false` - non-zero status and `nil` means that the execution failed).

Reply:
Sorry, I just saw we are raising an error in the other class in such a case, so scratch that.^